### PR TITLE
Issue 1: Updating enums

### PR DESCRIPTION
**Pull request by aislingvasey**
Multiple changes to Macaron project so please let me know what is correct and what should not be included please?

Changes to project:
- new dependencies: logging, mysql driver, midje (not used yet), etc (How did you run things without adding these dependencies such as driver?)
- added log4j.properties file to control logging but added it to test folder as I assume we wouldn't want to distribute with when any app is using macacon (exclude test folder in deploy?)
- removed old test file not in use I think and added test/macaron/example.clj for sample usage and testing.

Finally changes to core.clj to allow enums to be updated with additional types. Please can you have a look at functions and maybe suggest any changes to current implementation?

Thanks :)

### CODE REVIEW

File: project.clj [hunk 1]
```diff
@@ -2,4 +2,13 @@
   :description "A declarative entity abstraction library for clojure."
   :dependencies [[org.clojure/clojure "1.4.0"]
                  [org.clojure/tools.logging "0.2.3"]
-                 [org.clojure/java.jdbc "0.1.0"]])
\ No newline at end of file
+                 [org.clojure/java.jdbc "0.1.0"]
+                 [swank-clojure "1.4.2"]
```
Comment (CmdrDats) [inline project.clj]: The swank dependency needs to be in your ~/.lein/profiles.clj and not on this project as I wouldn't want this general use library to depend on swank. The same applies to the :dev profile below

File: test/macaron/example.clj [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+           (fields
+            [id :auto-index]
+            [version :version]
+            [uuid :unique-uuid :indexed]
+            [dateadded :datetime])
+           (post-create-fn macaron.core/add-entity-triggers))
+
+; A very simple types table
+(mc/defentity type1
+            (tablename "type1")
```
Comment (CmdrDats) [inline test/macaron/example.clj:58]: Just a note that if your tablename is the same as your entity name, you don't need to specify (tablename) - it's just if you have mysql being difficult with the name (like 'user'), you can specify another name ('users')

**Comment by CmdrDats** [comment]
Thank you! :)

**Comment by aislingvasey** [comment]
My pleasure! That was fun to do some real code :)

**Pull request merged**